playground/frontend/playground_components/lib/src/services/analytics/events/feedback_event_context.dart

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../../../models/event_snippet_context.dart';
+
+class FeedbackEventContext {
```

Comment:
Make `FeedbackController extends ChangeNotifier`, put to get_it.
Add `rating`, `EventSnippetContext`, and `additionalParams` to it.
Make TourNotifier update the feedback controller on unit change.
Make *PlaygroundNotifier update the feedback controller on `PlaygroundController` notifications.

Reply:
Couldn't test FeedbackController on PG. Not sure if because of incorrect configs or renderer selection.

<img width="241" alt="Screenshot 2023-04-07 at 20 45 42" src="https://user-images.githubusercontent.com/31556582/230628779-6cae499a-3b6c-40bb-a421-36e9d03bb68f.png">